packages/web-app-app-store/src/index.ts

```diff
@@ -0,0 +1,100 @@
+import { useGettext } from 'vue3-gettext'
+import { computed, unref } from 'vue'
+import {
+  AppMenuItemExtension,
+  defineWebApplication,
+  Extension,
+  useAbility,
+  useUserStore
+} from '@ownclouders/web-pkg'
+import { urlJoin } from '@ownclouders/web-client'
+import { RouteRecordRaw } from 'vue-router'
+import { useAppsStore, useRepositoriesStore } from './piniaStores'
+import { AppStoreConfigSchema } from './types'
+
+export const APPID = 'app-store'
+
+export default defineWebApplication({
+  setup({ applicationConfig }) {
+    const { $gettext } = useGettext()
+    const { can } = useAbility()
+    const userStore = useUserStore()
+    const repositoryStore = useRepositoriesStore()
+
+    const defaultRepositories = [
+      {
+        name: 'awesome-ocis',
+        url: 'https://raw.githubusercontent.com/owncloud/awesome-ocis/main/webApps/apps.json'
+      }
+    ]
+    if (applicationConfig?.repositories) {
+      const { repositories } = AppStoreConfigSchema.parse(applicationConfig)
+      repositoryStore.setRepositories(repositories || defaultRepositories)
+    } else {
+      repositoryStore.setRepositories(defaultRepositories)
+    }
+
+    const appInfo = {
+      name: $gettext('App Store'),
+      id: APPID,
+      icon: 'store',
+      color: '#ff6961'
+    }
+
+    const hasPermission = computed(() => {
+      // TODO: which permission(s) do we need to check here?
+      return userStore.user && can('read-all', 'Setting')
+    })
+
+    const routes: RouteRecordRaw[] = [
+      {
+        path: '/',
+        redirect: urlJoin(appInfo.id, 'list')
+      },
+      {
+        path: '/list',
+        name: 'list',
+        component: () => import('./views/AppList.vue'),
+        beforeEnter: (to, from, next) => {
+          if (!unref(hasPermission)) {
+            return next({ path: '/' })
+          }
+          next()
+        },
+        meta: {
+          authContext: 'user',
+          title: $gettext('App Store')
+        }
+      }
+    ]
+
+    const menuItemExtension: AppMenuItemExtension = {
+      id: `app.${appInfo.id}.menuItem`,
+      type: 'appMenuItem',
+      label: () => appInfo.name,
+      color: appInfo.color,
+      icon: appInfo.icon,
+      priority: 30,
+      path: urlJoin(appInfo.id)
+    }
+    const extensions = computed(() => {
+      const result: Extension[] = []
+
+      if (unref(hasPermission)) {
+        result.push(menuItemExtension)
+      }
+
+      return result
+    })
+
+    return {
+      appInfo,
+      routes,
+      extensions,
+      ready: () => {
+        const appsStore = useAppsStore()
+        return appsStore.loadApps()
+      }
```

Comment:
Why the ready hook? Why not ln AppsList?

Reply:
Because we need the apps loaded in every view/route. A wrapper for all views with `loadApps` in the `onMounted()` hook would be better probably...

Reply:
introduced a `LayoutContainer` component which wraps both the `list` and the `details` views and takes care of loading the apps before rendering the views.